src/core/aztec/encoder/CharMap.ts

```diff
@@ -5,26 +5,30 @@ import StringUtils from '../../common/StringUtils';
 import { int } from '../../../customTypings';
 
 export function static_CHAR_MAP(CHAR_MAP: Int32Array[]): Int32Array[] {
-  CHAR_MAP[C.MODE_UPPER][' '] = 1;
+  const spaceCharCode = StringUtils.getCharCode(' ');
+  const pointCharCode = StringUtils.getCharCode('.');
+  const commaCharCode = StringUtils.getCharCode(',');
+
+  CHAR_MAP[C.MODE_UPPER][spaceCharCode] = 1;
   const zUpperCharCode = StringUtils.getCharCode('Z');
   const aUpperCharCode = StringUtils.getCharCode('A');
   for (let c: int = aUpperCharCode; c <= zUpperCharCode; c++) {
     CHAR_MAP[C.MODE_UPPER][c] = c - aUpperCharCode + 2;
   }
-  CHAR_MAP[C.MODE_LOWER][' '] = 1;
+  CHAR_MAP[C.MODE_LOWER][spaceCharCode] = 1;
   const zLowerCharCode = StringUtils.getCharCode('z');
   const aLowerCharCode = StringUtils.getCharCode('a');
   for (let c: int = aLowerCharCode; c <= zLowerCharCode; c++) {
     CHAR_MAP[C.MODE_LOWER][c] = c - aLowerCharCode + 2;
   }
-  CHAR_MAP[C.MODE_DIGIT][' '] = 1;
+  CHAR_MAP[C.MODE_DIGIT][spaceCharCode] = 1;
   const nineCharCode = StringUtils.getCharCode('9');
   const zeroCharCode = StringUtils.getCharCode('0');
   for (let c: int = zeroCharCode; c <= nineCharCode; c++) {
     CHAR_MAP[C.MODE_DIGIT][c] = c - zeroCharCode + 2;
   }
-  CHAR_MAP[C.MODE_DIGIT][','] = 12;
-  CHAR_MAP[C.MODE_DIGIT]['.'] = 13;
+  CHAR_MAP[C.MODE_DIGIT][commaCharCode] = 12;
+  CHAR_MAP[C.MODE_DIGIT][pointCharCode] = 13;
```

Comment:
As much as I loved this change, I would ask you to not improve much the Java code, I would like to keep it as close as possible at least until we release v1.0.0. But this is fine.

Reply:
ok, it was really a bug fix as we compare char codes on some critical places.

Reply:
In Java its an int array as well, it just auto converts char to int, we need to do it manually.

Reply:
No problem, code looks nice.